### PR TITLE
Nasira custom item

### PR DESCRIPTION
**Pull request by Jaraci**
See title. Forum thread here: [url](https://forums.aurorastation.org/topic/17030-nasira-nahnikhs-incense-burner/#comment-155664)

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

**Review by Alberyk — APPROVED** [review]

File: code/modules/customitems/item_defines.dm [hunk 1]
```diff
@@ -2268,7 +2268,82 @@ All custom items with worn sprites must follow the contained sprite system: http
...
+/obj/item/fluff/nasira_burner/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/obj/item/fluff/nasira_burner/examine(mob/user)
+	..(user)
+	if(lit)
+		to_chat(user, "\The [src] is currently lit.")
+
+/obj/item/fluff/nasira_burner/proc/light(var/lighting_text = "[usr] lights \the [name].")
```
Comment (JohnWildkins) [inline code/modules/customitems/item_defines.dm:2296]:
```suggestion
/obj/item/fluff/nasira_burner/proc/light(var/lighting_text)
```
Just leave this undefined, using [usr] in a place that you don't need to should be avoided

File: code/modules/customitems/item_defines.dm [hunk 2]
```diff
@@ -2268,7 +2268,82 @@ All custom items with worn sprites must follow the contained sprite system: http
...
+	if(!lit)
+		lit = TRUE
+		playsound(src, 'sound/items/cigs_lighters/cig_light.ogg', 75, 1, -1)
+		var/turf/T = get_turf(src)
+		T.visible_message(SPAN_NOTICE(lighting_text))
```
Comment (JohnWildkins) [inline code/modules/customitems/item_defines.dm:2301]:
```suggestion
		if(lighting_text)
			var/turf/T = get_turf(src)
			T.visible_message(SPAN_NOTICE(lighting_text))
```

File: code/modules/customitems/item_defines.dm [hunk 3]
```diff
@@ -2268,7 +2268,82 @@ All custom items with worn sprites must follow the contained sprite system: http
...
+		set_light(2, 0.25, "#E38F46")
+		icon_state = "burner_lit"
+		START_PROCESSING(SSprocessing, src)
+
+/obj/item/fluff/nasira_burner/attack_self(mob/user as mob)
+	if(lit)
+		lit = FALSE
+		var/turf/T = get_turf(src)
+		T.visible_message(SPAN_NOTICE("[usr] extinguishes \the [src]."))
```
Comment (JohnWildkins) [inline code/modules/customitems/item_defines.dm:2310]:
```suggestion
		T.visible_message(SPAN_NOTICE("[user] extinguishes \the [src]."))
```
Don't use `usr` if you can avoid it

**Review by JohnWildkins — CHANGES_REQUESTED** [review]

**Review by JohnWildkins — APPROVED** [review]

**Pull request merged**